scripts/test/Muon/elemental_analysis_2/EA_grouping_tab/elemental_analysis_grouping_tab_presenter_test.py

```diff
@@ -0,0 +1,88 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from mantid.simpleapi import CreateSampleWorkspace
+from unittest import mock
+from mantidqt.utils.qt.testing import start_qapplication
+from mantid.api import WorkspaceGroup
+from qtpy.QtWidgets import QWidget
+
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_model import EAGroupingTabModel
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_widget_presenter import EAGroupingTabPresenter
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_view import EAGroupingTabView
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_table_widget_presenter import EAGroupingTablePresenter
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_table_widget_view import EAGroupingTableView
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+from Muon.GUI.Common.test_helpers.context_setup import setup_context_for_ea_tests
+
+
+@start_qapplication
+class EAGroupingTabPresenterTest(unittest.TestCase):
+    def setUp(self):
+        self.obj = QWidget()
+
+        self.loaded_data = MuonLoadData()
+
+        setup_context_for_ea_tests(self)
+
+        self.model = EAGroupingTabModel(context=self.context)
+
+        self.grouping_table_view = EAGroupingTableView()
```

Comment:
why are we not autospec-ing a mock? https://docs.python.org/3/library/unittest.mock.html#unittest.mock.create_autospec

Reply:
I had a lot of trouble making autospec work. Given the time constraints I stuck to what I could make work.